core/src/com/unciv/logic/automation/WorkerAutomation.kt

```diff
@@ -357,25 +363,23 @@ class WorkerAutomation(
 
             // Defence is more important that civilian improvements
             // While AI sucks in strategical placement of forts, allow a human does it manually
-            !civInfo.isPlayerCivilization() && evaluateFortPlacement(tile, civInfo, false) -> Constants.fort
+            !civInfo.isPlayerCivilization() && evaluateFortPlacement(tile, civInfo,false) -> Constants.fort
             // I think we can assume that the unique improvement is better
             uniqueImprovement != null && tile.canBuildImprovement(uniqueImprovement, civInfo)
                     && unit.canBuildImprovement(uniqueImprovement, tile) ->
                 uniqueImprovement.name
 
-            tile.terrainFeatures.contains("Fallout") -> "Remove Fallout"
-            tile.terrainFeatures.contains(Constants.marsh) -> "Remove Marsh"
+            lastTerrain.let {
+                isUnbuildableAndRemovable(it) && Automation.rankStatsValue(it, civInfo) < 0
+            } -> Constants.remove + lastTerrain.name
```

Comment:
Note that fallout doesn't have negative yield anymore, but instead has the unique "Nullifies all other stats this tile provides", so it won't be caught by this check

Reply:
Ayyyy good catch